execution/executor/src/lib.rs

```diff
@@ -516,14 +517,27 @@ where
             }
             outputs
         } else {
+            reexecuted_transactions = true;
             V::execute_block(transactions.clone(), &state_view)?
         };
 
         // Since other validators have committed these transactions, their status should all be
         // TransactionStatus::Keep.
-        for output in &vm_outputs {
-            if let TransactionStatus::Discard(_) = output.status() {
-                bail!("Syncing transactions that should be discarded.");
+        for (index, output) in vm_outputs.iter().enumerate() {
+            if let TransactionStatus::Discard(status_code) = output.status() {
+                let bail_error_message = format!(
+                    "Transaction version: {:?}\n
+                    Status code: {:?}\n
+                    Syncing transactions that have already been committed.",
+                    first_version, status_code
+                );
```

Comment:
nit: Can we also error log the bail message here, before we info log the rest? i.e.,

`error!(bail_error_message)`